core/src/main/scala/chisel3/CompileOptions.scala

```diff
@@ -22,6 +22,9 @@ trait CompileOptions {
   val explicitInvalidate: Boolean
   // Should the reset type of Module be a Bool or a Reset
   val inferModuleReset: Boolean
+
+  // If marked true, then any Module which consumes inferModuleReset=false must also mix in chisel3.RequireSyncReset
+  val migrateInferModuleReset: Boolean = false
```

Comment:
```suggestion
  def migrateInferModuleReset: Boolean = false
```
We really ought to default all virtual methods to def unless it truly is required to be a val. That being said I'm fine with keeping this as a `val` for consistency with the rest of the trait.